robot-db/src/main/java/dbconfiguration/SpringDBConfiguration.java

```diff
@@ -2,29 +2,35 @@
 
 
 import org.apache.commons.dbcp.BasicDataSource;
-import org.hibernate.SessionFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.PropertySource;
 import org.springframework.core.env.Environment;
-import org.springframework.orm.hibernate4.HibernateTransactionManager;
-import org.springframework.orm.hibernate4.LocalSessionFactoryBean;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.orm.jpa.vendor.Database;
+import org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter;
 
 import java.util.Properties;
 
 /**
  * The purpose of this class is to prepare and configure DB in Spring.
  */
 @Configuration
+@ComponentScan(basePackages = "repositories")
 @PropertySource("classpath:database.properties")
+@EnableJpaRepositories("dao")
 public class SpringDBConfiguration {
 
     @Autowired
     Environment env;
 
     /**
      * Bean to create BasicDataSource.
```

Comment:
No need for overly verbose Javadocs. Write important things, not names of objects you document, repeated or iterated.